### ISSUE TITLE
Optimistic local mutations with arrays

### ISSUE THREAD

**Issue by inzanez** [Kong/swrv#157]
Hi

I'm trying to use `mutate` with Vue3 `next` version to mutate the cache for a value that stores an array. As nothing worked so far, I had a look at `data.value` using `console.log`, and it seems that this is not stored as an array internally. I get some weird `Proxy` object with the array indices as properties.

Is there a way to actually use `mutate` for array values?

**Comment by darrenjennings** [comment]
> I get some weird Proxy object with the array indices as properties

`data` is a Vue [Ref](https://v3.vuejs.org/api/refs-api.html#ref)
https://github.com/Kong/swrv/blob/e7e5eaf931581119ca9f75ca4abdc2a43cfadf5f/src/types.ts#L27

**Comment by inzanez** [comment]
@darrenjennings yeah, I found out that much, but still had no luck mutating it :-)

**Comment by darrenjennings** [comment]
@inzanez do you have a codesandbox to repro? Here it is working as I expect (with vue composition api). Let me know if you think it is v3 specific.

https://codesandbox.io/s/swrv-157-79ek4

**Comment by inzanez** [comment]
@darrenjennings I just tried to reproduce it in a codesandbox, but I didn't manage. So I guess it might have something to do with the combination of quasar framework and vue3 or the specific setup that I'm using. I will try to figure it out, but in the meantime I can live well with awaiting 'revalidate' after a change.
